src/DynamoCoreWpf/ViewModels/PackageManager/PublishPackageViewModel.cs

```diff
@@ -540,6 +541,56 @@ public PublishPackageViewModel( DynamoViewModel dynamoViewModel ) : this()
             this.dynamoViewModel = dynamoViewModel;
         }
 
+        private void clearAllEntries()
+        {
+            // this function clears all the entries of the publish package dialog
+            this.Name = "";
+            this.RepositoryUrl = "";
+            this.SiteUrl = "";
+            this.License = "";
+            this.Keywords = "";
+            this.Description = "";
+            this.Group = "";
+            this.clearFullVersion();
+
+            this.Package = null;
+            this.Uploading = false;
+            this.UploadHandle = null;
+            this.IsNewVersion = false;
+            this.AdditionalFiles = new ObservableCollection<string>();
+            this.ErrorString = "";
+            this.MoreExpanded = false;
+            this.UploadState = PackageUploadHandle.State.Ready;
+
+            this.SubmitCommand.RaiseCanExecuteChanged();
+            this.PublishLocallyCommand.RaiseCanExecuteChanged();
+            this.ShowAddFileDialogAndAddCommand.RaiseCanExecuteChanged();
+            this.ToggleMoreCommand.RaiseCanExecuteChanged();
+            this.Dependencies = new ObservableCollection<PackageDependency>();
+            this.Assemblies = new List<PackageAssembly>();
+            this.clearPackageContents();
+        }
+
+        private void clearFullVersion()
+        {
+            this._MajorVersion = "";
+            RaisePropertyChanged("MajorVersion");
```

Comment:
When `_MajorVersion` is set, it automatically raises property changed event, so I dont think we should raise the event again. Please double-check its setter implementation.